matrix-connection.c

```diff
@@ -285,25 +302,12 @@ static void _join_failed(MatrixConnectionData *conn,
 
 
 void matrix_connection_join_room(struct _PurpleConnection *pc,
-        const gchar *room, GHashTable *components)
+        const gchar *room)
 {
-    GHashTable *copy;
-    GHashTableIter iter;
-    gpointer key, value;
-
     MatrixConnectionData *conn = purple_connection_get_protocol_data(pc);
 
-    /* we have to copy the components table, so that we can pass it back
-     * later on :/
-     */
-    copy = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
-    g_hash_table_iter_init (&iter, components);
-    while (g_hash_table_iter_next (&iter, &key, &value)) {
-        g_hash_table_insert(copy, g_strdup(key), g_strdup(value));
-    }
-
     matrix_api_join_room(conn, room, _join_completed, _join_error, _join_failed,
-            copy);
+            NULL);
```

Comment:
erm; passing NULL into user_data is going to make `_join_error` and `_join_failed` blow up